hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemCreate.java

```diff
@@ -146,6 +149,26 @@ public void testCreateNonRecursive2() throws Exception {
     assertIsFile(fs, testFile);
   }
 
+  @Test
+  public void testCreateOnRoot() throws Exception {
```

Comment:
AbstractContractRootDirectoryTest.testCreateFileOverRoot() shoud be doing some of this. 
1. are those tests running? that's ITestAbfsFileSystemContractRootDirectory
2. can you add the createNonRecursive test there?

Reply:
I checked and for some reason these tests are getting skipped even after adding the required configuration. Will debug more on this and get back.

Reply:
So, I checked further and found the tests were blocked because of the setting here: https://github.com/apache/hadoop/blob/trunk/hadoop-tools/hadoop-azure/src/test/resources/abfs.xml#L20

We have to set the required config fs.contract.test.root-tests-enabled as true here for these tests to run

Reply:
aah. well, do that then. it's off by default to stop people accidentally deleting their local disk and then complaining. (this has never happened, but...)

Reply:
Makes sense..
Set the config to true. 

Also, I was trying to add createNonRecursive in AbstractContractRootDirectoryTest. 
Looks like ContractTestUtils does not have support for this HDFS API. 
Directly getting File System instance and calling createNonRecursive could cause failure when this test wil run with file systems other than ABFS.

Reply:
good point. but at the same time, adding it will see what breaks...if you touch a file in the hdfs source tree (say its contract xml file...it can be left out the final commit) then we see what it does, and for s3a we will find out later and fix.